### ISSUE TITLE
Better error report when crate name is empty

### ISSUE THREAD

**Issue by orowith2os** [rust-lang/cargo#13314]
### Problem

When adding a crate via `cargo add`, if one adds a space before the version (such that it's `cargo add chumsky @1.0.0-alpha.6` instead of `cargo add chumsky@1.0.0-alpha.6` ), Cargo will panic with the following failed assertion:

```rust
    Updating crates.io index
thread 'main' panicked at src/cargo/core/dependency.rs:164:9:
assertion failed: !name.is_empty()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Proposed Solution

When the crate name to add is empty, print a nicer error instead.

### Notes

Here's another potentially interesting output to look at:

```rust
oro@CrystalBox /r/m/o/V/vala-thingy (master)> cargo add chumsky@ 1.0.0-alpha.6
error: invalid version requirement ``

Caused by:
  unexpected end of input while parsing major version number
  ```

**Comment by epage** [comment]
The panic has since been fixed.  Using nightly, I get
```
error: package name cannot be empty
```

It could be interesting to explore what are common cases that we can detect and provide better suggestions.  For example, as the person who merged `cargo add` into cargo, I recently did something like `cargo add winnow 0.5.13` (this started as a friendly jab about chumsky but I think it honestly was with winnow).
